week3/src/components/Card.jsx

```diff
@@ -0,0 +1,58 @@
+import styled from 'styled-components';
+// import theme from "./theme"
+
+
+export default function Card({ data, handleClick }) {
+    const { id, name, img, status, idx } = data
```

Comment:
지금 이 부분을 보면 상위 컴포넌트에서 구조분해 할당으로 data 객체를 받은 다음, data 객체를 또 구조분해 할당으로 unpacking 시켜주고 있어요. 상위 컴포넌트에서 객체 element를 넘겨줄 때는 spread 문법을 사용하면 좋아요!!

상위 컴포넌트에서 map을 순회하면서 각각의 e 객체를 data={e} 와 같이 넘겨주는 것을 확인했는데, 그냥 {...e} 처럼 spread 문법을 사용해서 넘겨준다면 이 부분에서 다음과 같이 코드를 간소화할 수 있어요!!

```suggestion
export default function Card({ id, name, img, status, idx, handleClick }) {
```

이렇게 해주면 상위 컴포넌트의 prop 이름에 하위 컴포넌트가 의존할 필요가 없습니다!!
뒤에서 한 번 더 설명드릴게요!!